addons/website_blog/static/src/js/website_blog.js

```diff
@@ -27,6 +28,21 @@ publicWidget.registry.websiteBlog = publicWidget.Widget.extend({
     // Handlers
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     * @param {Element} el
+     *
+     * Updates the href of an anchor tag when tags list is empty.
+     * This will redirect to backend part of the website blog post.
+     * TODO : Remove this in the master branch as it will be directly modified.
+     */
+    _onUpdateBackendHref(el) {
+        const id = window.location.href.match(/\/[\w-]+-(\d+)$/i)?.[1];
+        if (!id) {
+            return;
+        }
+        el.currentTarget.href = `/odoo/website/blog.post/${id}`;
```

Comment:
The blog ID is in the blog.name field in the DOM.
![image](https://github.com/user-attachments/assets/1a63e756-209d-4fce-9129-b74f86360aa2)
Get it from there, it's more reliable.
Your current method doesn't work if there's an anchor in the URL.
Also, update the href at the start, not in the click event.
(Why did you use the click event for that?)